### DISCUSSION TITLE
Pre-select nearest heading arguments

### DISCUSSION THREAD

**Discussion by csxtan** [minad/consult#939] [Q&A] [answered]
Not a bug in the main build, but in the helpful wiki, there is a [hack](https://github.com/minad/consult/wiki#pre-select-nearest-heading-for-consult-org-heading-and-consult-outline-using-vertico) for pre-selecting outline entries when using `consult-outline` by using advice. The function `consult--set-previous-point` is advised before `consult-outline` and `consult-org-heading` which have 1 and 2 optional arguments respectively. If these functions are called with arguments, [they get passed to the advice function](https://www.gnu.org/software/emacs/manual/html_node/elisp/Advice-Combinators.html) which does not accept arguments and throws "Wrong number of arguments".

This is easily resolved by changing the signature to:
```
(defun consult--set-previous-point (&optional arg1 arg2)
```

**Comment by minad** [comment]
Feel free to update the wiki if you have a working fix. Thanks!

**Reply by minad** [reply, marked as answer]
Fixed by @csxtan in https://github.com/minad/consult/wiki/Home/_compare/82084a2da1fb4870523b58ebf80974d5b00cd80b...d4802fcff8c4584967526964b19aefa73644d5df.
